### DISCUSSION TITLE
updating sequence doesn't update viewable

### DISCUSSION THREAD

**Discussion by ITwrx** [can-lehmann/owlkettle#51] [Q&A]
Whenever you have time, i was wondering if you might look at my CRUD attempt [here](https://source-code.itwrx.org/ITwrx/owlkettle_CRUD_PoC) 

Maybe you can easily see why my `list_loans_view` doesn't update? Specifically, on `line 35` of `/views/loans/list_loans_view` i add the new `Loan` to the `loans sequence`, but the `ListBox` starting on `ln 55` of same file doesn't update with the new data. 

I'm still using `owlkettle 2.2`, as i wanted to get this working before switching and trying out the Gtk.Calendar widget.

thanks

**Comment by can-lehmann** [comment]
The link to your code does not work for me, I get a 502 error when trying to visit it.

**Comment by ITwrx** [comment]
Sorry about that. It's up now. New server, that just happened to run some scripts and reboot for the first time while i was sleeping. Forgot to add gitea to runlevel default. :)

**Comment by can-lehmann** [comment]
The reason why the new loan does not get added to the list of loans is that, while you add the loan to the `LoanListView.loans` seq, it is not added to `App.loans` (seqs are passed by value in Nim). Since you set `loans = app.loans` when creating the `LoanListView`, any changes to the `LoanListView.loans` seq get replaced by whatever is in `App.loans`. I think you have 2 options to fix this:

1) Create a `LoanModel = ref object` to pass the loans by reference. (preferred option imo)
2) Use a callback to inform the `App` of the changes in `LoanListView.loans`.

**Reply by can-lehmann** [reply]
Ideally the loan model would also automatically handle all the database stuff so you don't have to call both `addNewLoanToLoans` and `loanListView.loans.add` in `LoanListView` (instead just `LoanModel.add`).

**Reply by ITwrx** [reply]
Thanks for the explanation. That cleared some things up for me. i _figured_ it had something to do with the fact that i stopped using the `ref object` you had in the examples i looked at. I was having trouble understanding how that was all working and how to make the `ref object` work in a modular/multi-file CRUD PoC. The `AddLoanDialog viewable` inputs were giving me trouble when using a `ref object`, IIRC.

I just sent you a fediverse DM, btw.
